cmd/agent/main.go

```diff
@@ -42,21 +42,24 @@ func sendMetrics(pollInterval time.Duration, host string) {
 	for {
 		time.Sleep(pollInterval * time.Second)
 
+		var tJSON []model.JSONMetrics
 		for k, v := range metrics.GetDataMetrics() {
-			sendRequestJSONFloat(host, "gauge", k, v.GetMetrics())
+			gauge := v.GetMetrics()
+			tJSON = append(tJSON, model.JSONMetrics{
+				ID:    k,
+				MType: "gauge",
+				Value: &gauge,
+			})
 		}
-		sendRequestJSONInt(host, "counter", count.GetName(), count.GetMetrics())
+		retrySendRequestJSONFloatAll(host, tJSON)
+		retrySendRequestJSONInt(host, "counter", count.GetName(), count.GetMetrics())
 		count.ClearCount()
```

Comment:
```suggestion
		err := retrySendRequestJSONFloatAll(host, tJSON)
		if err != nil {
			continue
		}
		err = retrySendRequestJSONInt(host, "counter", count.GetName(), count.GetMetrics())
		if err != nil {
			continue
		}
		count.ClearCount()
```
во всех случаях надо ловить (и, возможно, логировать) ошибку. Счетчик сбрасывается только в случае успешной отправки всех метрик. Ну или, как минимум, после успешной отправки самого счетчика.

Reply:
Исправлено

Reply:
+